//tensorflow/python/kernel_tests:init_ops_test fails on AVX512 builds

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): ('v1.8.0-14337-g18b1875', '1.13.0-rc0')
- Python version: Python 2.7.12
- Bazel version (if compiling from source): 0.19.2
- GCC/Compiler version (if compiling from source): gcc (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**
$  bazel test --config=opt  -- //tensorflow/python/kernel_tests:init_ops_test

//tensorflow/python/kernel_tests:init_ops_test                           FAILED in 1 out of 4 in 22.9s

**Describe the expected behavior**

The unit test should pass on AVX512 builds as it does on AVX2 builds.

**Code to reproduce the issue**

$ bazel test --config=opt  -- //tensorflow/python/kernel_tests:init_ops_test

**Other info / logs**
```
======================================================================
FAIL: testShapesValues (__main__.ConvolutionOrthogonal3dInitializerTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/user/.cache/bazel/_bazel_user/59a925eeb655b30b5c683f8317fe569e/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/init_ops_test.runfiles/org_tensorflow/tensorflow/python/framework/test_util.py", line 1103, in decorated
    return f(self, *args, **kwargs)
  File "/home/user/.cache/bazel/_bazel_user/59a925eeb655b30b5c683f8317fe569e/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/init_ops_test.runfiles/org_tensorflow/tensorflow/python/kernel_tests/init_ops_test.py", line 1142, in testShapesValues
    self.assertAllClose(self.evaluate(ratio), gain, rtol=tol, atol=tol)
  File "/home/user/.cache/bazel/_bazel_user/59a925eeb655b30b5c683f8317fe569e/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/init_ops_test.runfiles/org_tensorflow/tensorflow/python/framework/test_util.py", line 1013, in decorated
    return f(*args, **kwds)
  File "/home/user/.cache/bazel/_bazel_user/59a925eeb655b30b5c683f8317fe569e/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/init_ops_test.runfiles/org_tensorflow/tensorflow/python/framework/test_util.py", line 2121, in assertAllClose
    self._assertAllCloseRecursive(a, b, rtol=rtol, atol=atol, msg=msg)
  File "/home/user/.cache/bazel/_bazel_user/59a925eeb655b30b5c683f8317fe569e/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/init_ops_test.runfiles/org_tensorflow/tensorflow/python/framework/test_util.py", line 2090, in _assertAllCloseRecursive
    (path_str, path_str, msg)))
  File "/home/user/.cache/bazel/_bazel_user/59a925eeb655b30b5c683f8317fe569e/execroot/org_tensorflow/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/init_ops_test.runfiles/org_tensorflow/tensorflow/python/framework/test_util.py", line 2025, in _assertArrayLikeAllClose
    a, b, rtol=rtol, atol=atol, err_msg="\n".join(msgs), equal_nan=True)
  File "/home/user/.local/lib/python2.7/site-packages/numpy/testing/_private/utils.py", line 1452, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/home/user/.local/lib/python2.7/site-packages/numpy/testing/_private/utils.py", line 789, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=0.001, atol=0.001
Mismatched value: a is different from b. 
not close lhs = 3.15808677673
not close rhs = 3.14
not close dif = 0.0180867767334
not close tol = 0.00414
dtype = float32, shape = ()
(mismatch 100.0%)
 x: array(3.158087, dtype=float32)
 y: array(3.14)
```

--- a/tensorflow/core/kernels/eigen_contraction_kernel.h
+++ b/tensorflow/core/kernels/eigen_contraction_kernel.h
@@ -197,7 +197,8 @@ class TensorContractionBlocking<float, float, float, StorageIndex,
     // We split Kth dimensions in roughly equal slices.
     StorageIndex target_k_slices =
         (std::max)(StorageIndex(1), Eigen::divup(k, kc_));
-    StorageIndex packet_size = 8;
+    StorageIndex packet_size = internal::packet_traits<Scalar>::size;
+    if (packet_size < 8) packet_size = 8;
     StorageIndex target_bk =
         Eigen::divup(k / target_k_slices, packet_size) * packet_size;
     kc_ = (std::min)(k, target_bk);

